src/freak.jl

```diff
@@ -0,0 +1,105 @@
+type FREAK{S, T, O, N} <: DescriptorParams
+    pattern_scale::Float64
+    pattern_table::Array{S, N}
+    smoothing_table::Array{T, N}
+    orientation_weights::Array{O}
+end
+
+typealias OrientationPair Vector{Int}
+typealias SamplePair Vector{Float64}
+
+function FREAK(; pattern_scale::Float64 = 22.0)
+    pattern_table, smoothing_table = _freak_tables(pattern_scale)
+    orientation_weights = OrientationPair[]
+    for o in freak_orientation_sampling_pattern
+        offset_1 = pattern_table[1][o[1]]
+        offset_2 = pattern_table[1][o[2]]
+        dy, dx = offset_1 - offset_2
+        norm = (dx ^ 2 + dy ^ 2)
+        push!(orientation_weights, OrientationPair([round(Int, dy * 4096 / norm), round(Int, dx * 4096 / norm)]))
+    end
+    FREAK(pattern_scale, pattern_table, smoothing_table, orientation_weights)
+end
+
+function _freak_mean_intensity{T<:Gray}(int_img::AbstractArray{T, 2}, keypoint::Keypoint, offset::SamplePair, sigma::Float64)
+    y = k[1] + offset[1]
+    x = k[2] + offset[2]
+    if sigma < 0.5
+        tl = bilinear_interpolation(int_img, y - sigma, x - sigma)
```

Comment:
In the new ImagesFiltering, I think we should make `IntegralImage` a type and try to give it a nice interface. So that you could write this entire function something like this:
```jl
if σ < 0.5
    return interpolate(int_img, I±σ)
end
int_img[I±σ]
```
where `I` is a `CartesianIndex`.

This is pretty cute:
```jl
julia> ±(x, y) = (x-y, x+y)
± (generic function with 1 method)

julia> 5±3
(2,8)
```